sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -935,6 +935,105 @@ def sdc_pandas_dataframe_drop_impl(df, labels=None, axis=0, index=None, columns=
+                       f'  {result_c} = series_{c}.{func_name}()']
+        results.append((columns[i], result_c))
+
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f'  return pandas.DataFrame({{{data}}}{ind})']
+    func_text = '\n'.join(func_lines)
+    global_vars = {'pandas': pandas,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+"""
+Example func_text for func_name='isna' columns=('FLOAT', 'INT', 'STRING'):
+
+    def _df_isna_impl(df):
+        series_FLOAT = pandas.Series(get_dataframe_data(df, 0))
+        result_FLOAT = series_FLOAT.isna()
+        series_INT = pandas.Series(get_dataframe_data(df, 1))
+        result_INT = series_INT.isna()
+        series_STRING = pandas.Series(get_dataframe_data(df, 2))
+        result_STRING = series_STRING.isna()
+        return pandas.DataFrame({"FLOAT": result_FLOAT, "INT": result_INT, "STRING": result_STRING},
+                                index = df._index)
+"""
+
+
+def sdc_pandas_dataframe_isna_codegen(df, func_name, params, ser_params):
+    all_params = ['df']
+    df_func_name = f'_df_{func_name}_impl'
+    func_text, global_vars = _dataframe_reduce_columns_codegen_isna(func_name, all_params, df.columns, df)
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _reduce_impl = loc_vars[df_func_name]
+
+    return _reduce_impl
+
+
+def df_index_codegen_all(self):
+    if isinstance(self.index, types.NoneType):
+        func_lines = ''
+    else:
+        func_lines = ', index = df._index'
+
+    return func_lines
+
+
+@sdc_overload_method(DataFrameType, 'isna')
+def isna_overload(df):
+    """
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
+
+    Pandas API: pandas.DataFrame.isna
+
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/dataframe/dataframe_isna.py
+       :language: python
+       :lines: 35-
+       :caption: Detect missing values.
+       :name: ex_dataframe_isna
+
+    .. command-output:: python ./dataframe/dataframe_isna.py
+       :cwd: ../../../examples
+
+    .. seealso::
+
+        :ref:`DataFrame.isnull <pandas.DataFrame.isnull>`
+            Alias of isna.
+
+        :ref:`DataFrame.notna <pandas.DataFrame.notna>`
+            Boolean inverse of isna.
+
+        :ref:`DataFrame.dropna <pandas.DataFrame.dropna>`
+            Omit axes labels with missing values.
+
+        `pandas.absolute <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.isna.html#pandas.isna>`_
+            Top-level isna.
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+    Pandas DataFrame method :meth:`pandas.DataFrame.isna` implementation.
+
+    .. only:: developer
+        Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_df_isna*
+    """
+
+    return sdc_pandas_dataframe_isna_codegen(df, 'isna', {}, {})
```

Comment:
It would make sense to make params, ser_params have None value by default and just omit them here.
And inside the function redefine them if they are None as follows:
```
params = params if params is not None else {}
ser_params = ser_params if ser_params is not None else {}
```